app/Http/Controllers/Playlist/PlaylistsController.php

```diff
@@ -1190,11 +1195,57 @@ private function processMp3Audio($bible_files, $signed_files, $transaction_id, $
         return (object) ['hls_items' => $hls_items, 'signed_files' => $signed_files, 'durations' => $durations];
     }
 
+    /**
+     * The function returns the result of a comparison between the "verse_start" and "verse_end" properties of
+     * the "timestamp" that belongs to a transport_stream record. If they are not equal, the function returns
+     * true, indicating that the transport stream has a range of verses. If they are equal, the function returns
+     * false, indicating that the transport stream has a single verse.
+     *
+     * @param Collection $transport_stream
+     *
+     * @return bool
+     */
+    private function hasTransportStreamVerseRange(Collection $transport_stream) : bool
+    {
+        return $transport_stream->search(function ($stream) {
+            $timestamp = $stream->timestamp;
+            return $timestamp && (int)$timestamp->verse_start !== (int)$timestamp->verse_end;
```

Comment:
does this comparison work if $timestamp->verse_end is null?

Reply:
@bradflood , yes, it does, because if the $timestamp->verse_end value is null, it will be cast to 0